test/end2end_test.go

```diff
@@ -5509,6 +5510,131 @@ func (s) TestClientForwardsGrpcAcceptEncodingHeader(t *testing.T) {
 	}
 }
 
+func (s) TestServerHandlerSetSendCompressor(t *testing.T) {
```

Comment:
The reason for asking the tests to be split was not to reduce the number of test cases in each of them, but to simplify the test logic. So, what I was looking for was as follows:
- a test where all entries in the table are expected to succeed. So, the table here will not even have an entry for `wantErr`, and in the test logic you would call the function under test and fail the test if it returns an error.
- a test where all entries in the table are expected to fail. Here, the table will have an entry for `wantErr`, and in the test logic, you would fail the test if it did not return an error, or the returned error does not contain the string mentioned in `wantErr`.

The idea behind this suggestion is that the test logic is the one which should be more easily readable, and therefore it is OK to have some redundant code (in tests) in favor of readability.

Reply:
The refactoring I did earlier wasn't just to reduce the number of test cases, but rather split the test cases involving custom conditions such as conditionally calling the `sendHeader` API from the handler, disabling client advertisement, etc. into different tests and which I believe simplifies the testing logic. 

We just have 2 test cases that are expected to succeed, and one failure without any conditional setup. To help move this review faster, I'm going to go ahead and keep success test cases in one test which runs subtests for unary and stream. Another test case just for the failure, running subtest for unary and stream.